astropy/convolution/convolve.py

```diff
@@ -613,7 +613,10 @@ def convolve_fft(array, kernel, boundary='fill', fill_value=0.,
 
     # NaN and inf catching
     nanmaskarray = np.isnan(array) | np.isinf(array)
-    array[nanmaskarray] = 0
+    if nan_treatment == 'fill':
+        array[nanmaskarray] = fill_value
+    else:
+        array[nanmaskarray] = 0
```

Comment:
@jamienoss this line replaced the values with the default `fill_value` of zero.  So, the behavior was "always (kinda) correct" for `fill_value=0`, and we never had a test for other cases.

Reply:
(this comment is the answer to https://github.com/astropy/astropy/pull/8122#discussion_r233278804)

Reply:
Ok, cool, thanks. It was right in front of me after all.